backend/src/main/kotlin/org/jetbrains/kotlinconf/backend/services/SessionizeService.kt

```diff
@@ -0,0 +1,127 @@
+
+    private val sessionizeUrl: String = config.sessionizeUrl
+    private val sessionizeInterval: Long = config.sessionizeInterval
+    private val log = LoggerFactory.getLogger("SessionizeService")
+
+    @OptIn(DelicateCoroutinesApi::class)
+    private val syncJob = GlobalScope.launch {
+        log.info("Synchronizing each $sessionizeInterval minutes with $sessionizeUrl")
+        while (true) {
+            log.trace("Synchronizing to Sessionize…")
+            synchronizeWithSessionize(sessionizeUrl)
+            log.trace("Finished loading data from Sessionize.")
+            delay(TimeUnit.MINUTES.toMillis(sessionizeInterval))
+        }
+    }
+
+    suspend fun synchronizeWithSessionize(
+        sessionizeUrl: String,
+    ) {
+        val updatedValue = client.get(sessionizeUrl)
+            .body<SessionizeData>()
+            .toConference()
+
+        conference.emit(updatedValue)
+    }
+
+    suspend fun fetchImage(
+        imagesUrl: String,
+        imageId: String
+    ): ByteArray {
+        return client.get("$imagesUrl/$imageId").body<ByteArray>()
+    }
+
+    suspend fun getConferenceData(): Conference = conference.first()
+
+    private fun SessionizeData.toConference(): Conference {
+        val tags: Map<Int, CategoryItemData> = categories
+            .flatMap { it.items }
+            .associateBy { it.id }
+
+        fun findRoom(id: Int) = rooms.find { it.id == id }?.name ?: "unknown"
+
+        val sessions = sessions.mapNotNull { it ->
+            val startsAt = it.startsAt ?: return@mapNotNull null
+            val endsAt = it.endsAt ?: return@mapNotNull null
+            val tags = it.categoryItems.mapNotNull { tags[it]?.name }
+            Session(
+                it.id,
+                it.displayTitle,
+                it.descriptionText ?: "",
+                it.speakers,
+                it.roomId?.let { findRoom(it) } ?: "unknown",
+                startsAt,
+                endsAt,
+                tags
+            )
+        }.mergeWorkshops()
+
+        val speakers = speakers.map {
+            Speaker(
+                it.id,
+                it.fullName,
+                it.tagLine ?: "",
+                it.bio ?: "",
+                it.profilePicture ?: ""
+            )
+        }
+
+        return Conference(sessions, speakers)
+    }
+
+    private fun List<Session>.mergeWorkshops(): List<Session> {
+        val (workshopParts, nonWorkshop) = partition { it.tags?.contains("Workshop") == true }
+
+        // Group by room
+        val workshopPartsByLocations = workshopParts.groupBy { it.location }
+        val workshops = workshopPartsByLocations.values.mapNotNull { parts ->
+            val startTime = parts.minOf { it.startsAt }
+            val endTime = parts.maxOf { it.endsAt }
+            val first = parts.find { it.title.contains("Part 1") } ?: return@mapNotNull null
+            val title = first.title.substringAfter(". Part").trim()
```

Comment:
This currently returns titles like `"title": "1."`. We need the part of the title before the part indicator.

```suggestion
            val title = first.title.substringBefore(". Part").trim()
```

Reply:
Some of the titles :( 

Some of them keep part 1: Spring Boot With Coroutines and Virtual Threads. Part 1.